src/main/groovy/com/vherasymenko/avro/decoder/core/AvroDecoderService.groovy

```diff
@@ -0,0 +1,25 @@
+package com.vherasymenko.avro.decoder.core
+
+import event.course_install.CourseInstall
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DecoderFactory
+
+/**
+ * The avro decoder.
+ */
+class AvroDecoderService implements AvroDecoderPort {
+
+    @Override
+    void decodeEvent( String event ) {
+        def inputStream = new ByteArrayInputStream( event.bytes )
```

Comment:
+2, you can get very easily burned by not explicitly stating your encoding. It's happened several times to us and the TLO team.